Added possibility to have many Sentry instances and different http request mechanism

lsuski opened the pull request:
I've made some refactorings to achieve my requirements. Firstly I didn't want to use deprecated apache client and I needed some security configuration of http request sending. Secondly I needed to use Sentry in library and to avoid any conflicts I've added possibility to have many Sentry instances. Finally I've made some optimisations, e.g. writing requests as JSON because Java Serialization is very unoptimal on Android, especially that Sentry already used JSON. I've also disabled logging for default - it can be turned on by calling `setDebugLogging` method.
Review carefully my changes.
Thanks

marcomorain commented:
Hi @lsuski - thanks for the contribution. Sorry about not getting back to you sooner.

This diff is pretty large. I'd like to merge the features, but I think it's safer (and much easier to review) if we implement the changes in smaller chunks. I have implemented a similar change to yours, to use an executor service already.

I'm going to open a GitHub issues to track the individual changes here.

lsuski commented:
I've merged changes from your master branch to my fork. I've changed the way of sending contexts - this functionality is not default now and is available through `AppInfoSentryCaptureListener`. I've changed it because I don't need so many information about device and I have different release versioning beacuse I use Sentry in library. I've also fixed the way of passing credentials which causes Google Play warning.
